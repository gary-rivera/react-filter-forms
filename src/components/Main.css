.main-container {
  box-shadow: rgba(0, 0, 0, 0.12) 0px 2px 5px, rgba(0, 0, 0, 0.24) 0px 2px 3px;
  background: #f8f8ff;
  border-radius: 10px;
  width: 60vw;
  height: 80vh;
  margin: auto;

  overflow-y: auto;
}

.main-container::-webkit-scrollbar {
  display: none;
}
