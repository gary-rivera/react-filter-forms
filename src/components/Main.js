import './Main.css';
import { useState } from 'react';
import FilterForm from './FilterForm';
import Students from './Students';

/** Main container for filtering forms and students
 *
 * Props:
 * - students: array of { student, ... }
 *
 * State:
 * - filters: object with a filter for each input 'name' and 'tag' that we could potentially filter by
 *
 * {App} -> Main -> { FilterForm }
 *               -> { Students }
 */

function Main({ students }) {
  const [filters, setFilters] = useState({
    "name": "",
    "tag": ""
  });

  // helper function FilterForm uses to pass up form data to filters state.
  function addFilter(filter){
    setFilters(filter);
  }

  return (
    <div className="main-container">
      <FilterForm
        filters={filters}
        addFilter={addFilter}
      />
      <Students students={students} filters={filters}/>
    </div>
  )
}

export default Main;