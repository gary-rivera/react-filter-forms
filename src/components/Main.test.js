import { render } from '@testing-library/react';
import Main from './Main';

const students = [
  {
    city: "SF",
    company: "Google", 
    email: "user1@domain.com", 
    firstName: "user1", 
    grades: ['0', '100'],
    id: "1",
    lastName: "user1",
    pic: "https://as1.ftcdn.net/jpg/03/46/83/96/500_F_346839683_6nAPzbhpSkIpb8pmAwufkC7c5eD7wYws.jpg",
    skill: "JS",
    tags: ['tag1', 'tag2', 'tag3']
  }, 
  {
    city: "LA",
    company: "Google", 
    email: "user2@domain.com", 
    firstName: "user2", 
    grades: ['0', '0', '0'],
    id: "2",
    lastName: "user2",
    pic: "https://as1.ftcdn.net/jpg/03/46/83/96/500_F_346839683_6nAPzbhpSkIpb8pmAwufkC7c5eD7wYws.jpg",
    skill: "JS",
    tags: []
  }, 
]
it('renders without crashing', function () {
  render(<Main students={students} />);
});

it("renders the correct amount of students", () => {
  const { container } = render(<Main students={students} />)
  const studentCards = container.querySelectorAll('.student-card')
  expect(studentCards.length).toEqual(2);
});

it("matches snapshot", function () {
  const { container } = render(<Main students={students} />);
  expect(container).toMatchSnapshot();
});