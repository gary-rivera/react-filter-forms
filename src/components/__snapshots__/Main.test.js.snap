// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="main-container"
  >
    <form
      class="filter-form"
    >
      <input
        autocomplete="off"
        for="name"
        label="name"
        name="name"
        placeholder="Search by name"
        value=""
      />
      <input
        autocomplete="off"
        for="tag"
        label="tag"
        name="tag"
        placeholder="Search by tag"
        value=""
      />
    </form>
    <div
      class="student-card"
    >
      <div
        class="profile-pic"
      >
        <img
          alt="user1 user1"
          src="https://as1.ftcdn.net/jpg/03/46/83/96/500_F_346839683_6nAPzbhpSkIpb8pmAwufkC7c5eD7wYws.jpg"
        />
      </div>
      <div
        class="student-info"
      >
        <h1>
          USER1 USER1
        </h1>
        <div
          class="details"
        >
          <h4
            class="student-email"
          >
            Email: 
            user1@domain.com
          </h4>
          <h4
            class="student-company"
          >
            Company: 
            Google
          </h4>
          <h4
            class="student-skill"
          >
            Skill: 
            JS
          </h4>
          <h4
            class="student-average-score"
          >
            Average: 
            50
            %
          </h4>
          <ul
            class="hidden"
          >
            <li>
              Test 
              1
              :
              <span>
                0
                %
              </span>
            </li>
            <li>
              Test 
              2
              :
              <span>
                100
                %
              </span>
            </li>
          </ul>
          <div
            class="tags-container"
          >
            <div
              class="tag-icons"
            >
              <div
                class="tag"
              >
                 
                tag1
              </div>
              <div
                class="tag"
              >
                 
                tag2
              </div>
              <div
                class="tag"
              >
                 
                tag3
              </div>
            </div>
            <form>
              <input
                autocomplete="off"
                class="tag-form"
                name="addTag"
                placeholder="Add a tag"
                type="text"
                value=""
              />
            </form>
          </div>
        </div>
      </div>
      <button>
        +
      </button>
    </div>
    <div
      class="student-card"
    >
      <div
        class="profile-pic"
      >
        <img
          alt="user2 user2"
          src="https://as1.ftcdn.net/jpg/03/46/83/96/500_F_346839683_6nAPzbhpSkIpb8pmAwufkC7c5eD7wYws.jpg"
        />
      </div>
      <div
        class="student-info"
      >
        <h1>
          USER2 USER2
        </h1>
        <div
          class="details"
        >
          <h4
            class="student-email"
          >
            Email: 
            user2@domain.com
          </h4>
          <h4
            class="student-company"
          >
            Company: 
            Google
          </h4>
          <h4
            class="student-skill"
          >
            Skill: 
            JS
          </h4>
          <h4
            class="student-average-score"
          >
            Average: 
            0
            %
          </h4>
          <ul
            class="hidden"
          >
            <li>
              Test 
              1
              :
              <span>
                0
                %
              </span>
            </li>
            <li>
              Test 
              2
              :
              <span>
                0
                %
              </span>
            </li>
            <li>
              Test 
              3
              :
              <span>
                0
                %
              </span>
            </li>
          </ul>
          <div
            class="tags-container"
          >
            <form>
              <input
                autocomplete="off"
                class="tag-form"
                name="addTag"
                placeholder="Add a tag"
                type="text"
                value=""
              />
            </form>
          </div>
        </div>
      </div>
      <button>
        +
      </button>
    </div>
  </div>
</div>
`;
